posthog/auth.py

```diff
@@ -143,8 +143,13 @@ def authenticate(self, request: Request):
             if not user.exists():
                 raise AuthenticationFailed(detail="User doesn't exist")
             return (user.first(), None)
+
         return None
 
+    # NOTE: This appears first in the authentication chain often so we want to define an authenticate_header to ensure 401 and not 403
```

Comment:
I don't know all the details behind this, but it might just be cleaner and add no new lines to subclass from `TokenAuthentication` which provides this method.

Reply:
I also don't know the detail and would rather not make that change here. Happy to review a follow up PR though!

Reply:
Also - we would override almost the entire base class, only inherting a couple of properties. Doesn't feel worth it imo